From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Sat, 8 Aug 2026 17:48:05 +0200
Subject: [PATCH] Add groups.d definition for the webapplication.push ACG

com.palm.webappmanager.api.json puts
com.webos.service.webappmanager/fireNotificationEvent in the
webapplication.push access control group, but
com.webos.service.webappmanager.groups.json only declares
webapplication.management:

WARNING: luneos-dev-image-1.0-r0 do_validate_ls2_acg: Found 2 group(s) that appear only in api-permissions.d, consider define them in groups.d
=== LIST BEGIN: Groups used in api-permissions.d but not defined in groups.d ===
  webapplication.push
=== LIST END ===

Declare the group and add com.webos.service.webappmanager to
allowedNames, since that is the name the API is registered under (it is
already listed in com.palm.webappmanager.role.json.in).
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Pending
---
 files/sysbus/com.webos.service.webappmanager.groups.json | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/files/sysbus/com.webos.service.webappmanager.groups.json b/files/sysbus/com.webos.service.webappmanager.groups.json
index 98cff69..e5c49a6 100644
--- a/files/sysbus/com.webos.service.webappmanager.groups.json
+++ b/files/sysbus/com.webos.service.webappmanager.groups.json
@@ -1,4 +1,5 @@
 {
-	"allowedNames" : ["com.palm.webappmanager"],
-	"webapplication.management" : ["oem"]
+	"allowedNames" : ["com.palm.webappmanager", "com.webos.service.webappmanager"],
+	"webapplication.management" : ["oem"],
+	"webapplication.push" : ["oem"]
 }
